docs/connectors.md

```diff
@@ -0,0 +1,343 @@
+- Uses `SubjectSourceBackgroundService` for lifecycle management
+
+**Cardinality**: Each property can have at most one source (single source of truth)
+
+**Examples**: OPC UA client connecting to a PLC, database client, REST API consumer
+
+### ISubjectSource Interface
+
+```csharp
+public interface ISubjectSource : ISubjectConnector
+{
+    /// <summary>
+    /// Checks whether the property is handled by this source.
+    /// Implementations typically delegate to an internal IPathProvider.
+    /// </summary>
+    bool IsPropertyIncluded(RegisteredSubjectProperty property);
+
+    /// <summary>
+    /// Maximum batch size for write operations (0 = no limit).
+    /// </summary>
+    int WriteBatchSize { get; }
+
+    /// <summary>
+    /// Start listening for external changes.
+    /// </summary>
+    Task<IDisposable?> StartListeningAsync(
+        SubjectPropertyWriter propertyWriter,
+        CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Write property changes back to the external system.
+    /// </summary>
+    ValueTask WriteChangesAsync(
+        ReadOnlyMemory<SubjectPropertyChange> changes,
+        CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Load initial state from the external system.
+    /// </summary>
+    Task<Action?> LoadInitialStateAsync(CancellationToken cancellationToken);
+}
+```
+
+### Data Flow
+
+#### Inbound (External → Subject)
+
+When the external system sends new values:
+
+1. External System sends update
+2. Source receives the update
+3. Source calls `propertyWriter.Write()`
+4. Subject property is updated
+
+#### Outbound (Subject → External)
+
+When you change a property value in code:
+
+1. Subject property is changed
+2. `WriteChangesAsync()` is called with the change
+3. Source sends update to external system
+
+### Initialization Sequence
+
+Sources use a buffer-flush-load-replay pattern during initialization and reconnection:
+
+1. **Buffer**: During `StartListeningAsync()`, inbound updates are buffered
+2. **Flush**: Pending writes from the retry queue are flushed to the external system
+3. **Load**: `LoadInitialStateAsync()` fetches complete state from external system
+4. **Replay**: Buffered updates are replayed in order after initial state is applied
+
+This ensures:
+- Updates received during initialization are not lost
+- Updates are applied in the correct order relative to the initial state
+- No visible state toggle (flush before load avoids showing stale server values)
+
+### Inbound Update Error Handling
+
+When applying inbound updates (writing data from the external system to the local subject model), if an individual update fails (the action throws an exception), the error is logged and **the update is dropped**. There is no retry mechanism for inbound updates.
+
+This is by design:
+- Individual update failures don't block other updates from being applied
+- Monitor logs for `Failed to apply subject update` errors to detect issues
```

Comment:
maybe add: - write failures on internal models are considered non-transient because a regular property write should just work or will not work on retry anyway (that's why we do not retry and just log), even if there is a custom validation failure the assumption is that it's non-transient